### ISSUE TITLE
Failing test-fileio in docker

### ISSUE THREAD

**Issue by bobrik** [systemd/systemd#27175]
### systemd version the issue has been seen with

253

### Used distribution

Debian

### Linux kernel version used

6.1.14

### CPU architectures issue was seen on

x86_64

### Component

tests

### Expected behaviour you didn't see

I expect all tests to pass in docker.

### Unexpected behaviour you saw

We see `fileio` test fail:

```
 565/1094 systemd:test / test-fileio                                           FAIL           0.08s (killed by signal 6 SIGABRT)

--- command ---
23:15:39 SYSTEMD_LANGUAGE_FALLBACK_MAP='/cfsetup_build/build/systemd/src/locale/language-fallback-map' PROJECT_BUILD_ROOT='/cfsetup_build/build/systemd/obj-x86_64-linux-gnu' PATH='/cfsetup_build/build/systemd/obj-x86_64-linux-gnu:/usr/sbin:/usr/bin:/sbin:/bin:/usr/bin/X11' /cfsetup_build/build/systemd/obj-x86_64-linux-gnu/test-fileio
--- stderr ---
/* test_test_read_virtual_file */
/* test_read_virtual_file_one (max_size=0) */
read_virtual_file("/proc/1/cmdline", 0): truncated (0 bytes)
read_virtual_file("/etc/nsswitch.conf", 0): truncated (0 bytes)
read_virtual_file("/sys/kernel/uevent_seqnum", 0): truncated (0 bytes)
read_virtual_file("/proc/kcore", 0): Bad file descriptor
Assertion 'ERRNO_IS_PRIVILEGE(r) || IN_SET(r, -ENOENT, -EINVAL, -EFBIG)' failed at src/test/test-fileio.c:1023, function test_read_virtual_file_one(). Aborting.
-------
```

This is due to the fact that docker masks `/proc/kcore`, which makes it a char device inside of the container:

```
$ stat /proc/kcore
  File: /proc/kcore
  Size: 0         	Blocks: 0          IO Block: 4096   character special file
Device: 7ch/124d	Inode: 5           Links: 1     Device type: 1,3
Access: (0666/crw-rw-rw-)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-04-06 23:17:12.975071539 +0000
Modify: 2023-04-06 23:17:12.975071539 +0000
Change: 2023-04-06 23:17:12.975071539 +0000
 Birth: 2023-04-06 23:17:12.975071539 +0000
```

Compare to the host:

```
$ stat /proc/keys
  File: /proc/keys
  Size: 0         	Blocks: 0          IO Block: 1024   regular empty file
Device: 12h/18d	Inode: 4026532142  Links: 1
Access: (0444/-r--r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-04-06 23:26:19.994602860 +0000
Modify: 2023-04-06 23:26:19.994602860 +0000
Change: 2023-04-06 23:26:19.994602860 +0000
 Birth: -
```

Here's the check that fails in systemd:

* https://github.com/systemd/systemd/blob/v253/src/basic/fileio.c#L440-L441

You can find the list of masked paths here:

* https://github.com/moby/moby/blob/v23.0.3/oci/defaults.go#L90-L101

### Steps to reproduce the problem

Run the tests in a docker container.

**Comment by bobrik** [comment]
`/proc/kcore` is already special in tests:

```c
assert_se(ERRNO_IS_PRIVILEGE(r) || /* /proc/kcore is not accessible to unpriv */
       IN_SET(r,
              -ENOENT,  /* Some of the files might be absent */
              -EINVAL,  /* too small reads from /proc/self/pagemap trigger EINVAL */
              -EFBIG)); /* /proc/kcore and /proc/self/pagemap should be too large */
```

**Comment by yuwata** [comment]
Fix is waiting in #27176.
